src/mono/wasm/runtime/library_mono.js

```diff
@@ -1507,6 +1508,12 @@ var MonoSupportLib = {
 			Module.ccall ('mono_wasm_load_profiler_coverage', null, ['string'], [arg]);
 		},
 
+		mono_wasm_register_custom_marshaler: function (aqn, marshalerAQN) {
```

Comment:
In tests and potentially blazor, it's for cases where you don't control mono_config. The benchmarks use it right now to register their marshalers

Reply:
Like so:

```csharp
#if NEWAPI
        Interop.Runtime.InvokeJS($"MONO.mono_wasm_register_custom_marshaler('System.Uri', 'System.Runtime.InteropServices.JavaScript.UriMarshaler')", out temp);
```

We could provide a built in method to register them all, but the long term plan is to be able to link them out, so any consumer not using mono_config needs to feed that information into the runtime themselves when initializing it

Reply:
Resolving because I don't see any obvious way to improve on this, and I will be explaining it clearly in the documentation. (It's also used in the benchmarks that will get added to dotnet/performance eventually)